src/tests/snapshots/gitu__tests__discard__discard_file_move.snap

```diff
@@ -19,7 +19,7 @@ expression: ctx.redact_buffer()
                                                                                 |
                                                                                 |
                                                                                 |
-                                                                                |
 ────────────────────────────────────────────────────────────────────────────────|
-$ git checkout HEAD -- new-file                                                 |
-styles_hash: 19cf9a9796ec3951
+$ git rm --force moved-file                                                     |
+rm 'moved-file'                                                                 |
+styles_hash: bc84efbf82da3787
```

Comment:
I'm not sure what this is supposed to mean and if it's expected. Any idea @altsem ?

Reply:
The "discard" module runs different git commands depending on context.
https://github.com/altsem/gitu/blob/86603c81fd015fc16d25c76757b99a87522a34fc/src/ops/discard.rs#L14-L17

It just doesn't properly handle moved files yet.
I tested this out in Magit now to see how it handles it, it seems:

Unstaged:
![image](https://github.com/user-attachments/assets/d415064d-795f-46be-9cc0-c28c9a603977)

Staged:
![image](https://github.com/user-attachments/assets/e9f190d1-9322-478d-a4f1-6cd83cf001a6)

And discarding:
![image](https://github.com/user-attachments/assets/f7336b8c-af15-437f-9643-fe7e34a0181f)

All resulted in:
![image](https://github.com/user-attachments/assets/e9d99708-623a-43a3-ab4a-39199c0599f9)

Reply:
Ah, I see. If I get that right, I need to change the discard module to run something like the following in case of renames, correct?

```
git mv new_file old_file
```

Reply:
I think so!

Reply:
Fixed. The discard command now handles renames based on my testing.

Reply:
Testing this again. If there are other changes in addition to renaming a file those changes won't get reverted. 

I think we need to run the checkout command in addition to `rename_file`

Reply:
I never used the original `magit` so I'm not how it behaves in that case. Does it require two discards (first for the rename and another for the content changes)?

Reply:
Interesting, I'd test this out in Magit and just copy the behaviour if possible.

I realised there may be other OpTrait implementations that may bork with this new "renamed" status.

The trait is found here:
https://github.com/altsem/gitu/blob/master/src/ops/mod.rs#L30

It's in the "TargetData" parameter.
When the TargetData is of the "Delta" variant.

I'd help test things out in Magit but I'm off the computer for a few days.

Reply:
I just installed Emacs for the first time lol. Magit does one revert at a time. First, it reverts the renaming, and then you have to press `k` again to revert the other changes in the file. That matches the current behavior in `gitu` so I think we are good.